spec/ama_layout/notification_set_spec.rb

```diff
@@ -0,0 +1,251 @@
+describe AmaLayout::NotificationSet do
+  let(:store) do
+    AmaLayout::Notifications::RedisStore.new(
+      db: 4,
+      namespace: 'test_notifications',
+      host: 'localhost'
+    )
+  end
+  let(:key) { 1 }
+  let(:duration) { AmaLayout::Notification::DEFAULT_LIFESPAN.to_i }
+  let(:store_key) { key.to_s }
+  let(:json) do
+    <<-JSON
+    {
+      "8ca9f850c18acc17643038b2341bee3ede8a24c0f3e92f56f2109ce49fdcb616": {
+        "type": "notice",
+        "header": "test",
+        "content": "test",
+        "created_at": "2017-06-19T06:00:00.000Z",
+        "active": true,
+        "lifespan": #{duration},
+        "version": "1.0.0"
+      }
+    }
+    JSON
+  end
+  let(:stale_json) do
+    <<-JSON
+    {
+      "d3c2bc71904100674325791b371db7446097f956ea76a304e787abd5f2588665": {
+        "type": "notice",
+        "header": "stale",
+        "content": "stale",
+        "created_at": "2012-06-19T06:00:00.000Z",
+        "active": true,
+        "lifespan": #{duration},
+        "version": "1.0.0"
+      }
+    }
+    JSON
+  end
+
+  subject { described_class.new(store, key) }
+
+  around(:each) do |example|
+    Timecop.freeze(Time.zone.local(2017, 6, 19)) do
+      store.clear
+      example.run
+      store.clear
+    end
+  end
+
+  describe '#intialize' do
+    context 'with valid JSON in data store' do
+      before(:each) do
+        store.set(store_key, notification)
+      end
+
+      context 'without stale notifications in the data store' do
+        let(:notification) { json }
+
+        it 'fetches the notifications' do
+          expect(subject.size).to eq(1)
+        end
+      end
+
+      context 'with stale notifications in the data store' do
+        let(:notification) { stale_json }
+
+        it 'returns an empty set' do
+          expect(subject).to be_empty
+        end
+
+        it 'cleans out stale notifications from the data store' do
+          subject
+          expect(store.get(store_key)).to eq('{}')
+        end
+      end
+    end
+
+    context 'with invalid JSON in data store' do
+      before(:each) do
+        store.set(store_key, '{"invalid_json":')
+      end
+
+      it 'logs to Rails logger' do
+        expect(Rails.logger).to receive(:error).with(instance_of(String))
+        subject
+      end
+
+      it 'deletes the key in data store' do
+        subject
+        expect(store.get(store_key)).to be nil
+      end
+
+      it 'returns an empty set' do
+        expect(subject).to be_empty
+      end
+
+      it 'sets the base attribute to a hash' do
+        expect(subject.base).to be_a(Hash)
+      end
+    end
+
+    context 'with no entry in data store' do
+      it 'returns an empty set' do
+        expect(subject).to be_empty
+      end
+    end
+  end
+
+  describe '#each' do
+    before(:each) do
+      store.set(store_key, json)
+    end
+
+    it 'iterates the array' do
+      output = []
+      subject.each do |element|
```

Comment:
Could you just use `subject.to_a` instead of `output`, since the NotificationSet class includes enumerable?